zulipterminal/helper.py

```diff
@@ -816,3 +817,61 @@ def open_media(controller: Any, tool: str, media_path: str) -> None:
 
     if error:
         controller.report_error(error)
+
+
+def user_friendly_time(time_stamp: str) -> str:  # message_info has a different format
+
+    now = datetime.now()
+
+    if len(time_stamp) > 22:  # for message_info, year is included
+        datetime_object = datetime.strptime(
+            time_stamp, "%a %b %d %Y %I:%M:%S %p"
+        )  # Fri Apr 29 2022 05:15:12 AM is an example
+    elif len(time_stamp) > 0:  # for user_info, year not included
+        if time_stamp[-1] == "M":  # AM or PM
+            datetime_object = datetime.strptime(
+                time_stamp + str(now.year), "%a %b %d %I:%M:%S %p%Y"
+            )  # Fri Apr 29 05:15:12 AM is an example
+        else:
+            datetime_object = datetime.strptime(
+                time_stamp + str(now.year), "%a %b %d %H:%M:%S%Y"
+            )  # Fri Apr 29 15:15:12 is an example
+    else:
+        return ""  # return blank string if input is "" or similar
+
+    time_spent = now - datetime_object
+
+    seconds = time_spent.seconds
+    days = time_spent.days
+
+    if days < 0 or seconds < -30:
+        return ""
+
+    if days == 0:
+        if seconds < 10:
+            return "Just now"
+        if seconds < 60:
+            return str(seconds) + " seconds ago"
+        if seconds < 120:
+            return "A minute ago"
+        if seconds < 3600:
+            return str(seconds // 60) + " minutes ago"
+        if seconds < 7200:
+            return "An hour ago"
+        if seconds < 86400:  # 24 hours
+            return str(seconds // 3600) + " hours ago"
+
+    time_string = ""
```

Comment:
The value of `time_string` is assigned only in the first conditional, but not returned.